superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterSets/FilterSetUnit.test.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import { mockStore } from 'spec/fixtures/mockStore';
+import { Provider } from 'react-redux';
+import userEvent from '@testing-library/user-event';
+import FilterSetUnit, { FilterSetUnitProps } from './FilterSetUnit';
+
+const mockedProps = {
+  editMode: true,
+  setFilterSetName: jest.fn(),
+  onDelete: jest.fn(),
+  onEdit: jest.fn(),
+  onRebuild: jest.fn(),
+};
+
+function openDropdown() {
+  const dropdownIcon = screen.getByRole('img', { name: 'ellipsis' });
+  userEvent.click(dropdownIcon);
+}
+
+const setup = (props: FilterSetUnitProps) => (
+  <Provider store={mockStore}>
+    <FilterSetUnit {...props} />
+  </Provider>
+);
+
+test('should render', () => {
+  const { container } = render(setup(mockedProps));
+  expect(container).toBeInTheDocument();
+});
+
+test('should render the edit button', () => {
+  const editModeOffProps = {
+    ...mockedProps,
+    editMode: false,
+  };
+  render(setup(editModeOffProps));
+  const editButton = screen.getAllByRole('button')[0];
```

Comment:
Can you use `getByRole('button', { name: 'Edit' })`? If not, can you use `getByLabelText('Edit')`?